napari/utils/_magicgui.py

```diff
@@ -66,21 +80,29 @@ def register_types_with_magicgui():
         layers.Layer, choices=get_layers, return_callback=add_layer_to_viewer
     )
     register_type(Viewer, bind=find_viewer_ancestor)
-    register_type(
-        types.LayerDataTuple,
-        return_callback=add_layer_data_tuples_to_viewer,
-    )
-    register_type(
-        List[types.LayerDataTuple],
-        return_callback=add_layer_data_tuples_to_viewer,
-    )
+
+    for _type in (types.LayerDataTuple, List[types.LayerDataTuple]):
+        register_type(_type, return_callback=add_layer_data_tuples_to_viewer)
+        register_type(Future[_type], return_callback=add_future_data)  # type: ignore
```

Comment:
Future is subscript-able only on 3.9+

Reply:
Yeah I realized that later :/ have to go back to the original idea... or ditch it